libraries/core_libs/consensus/src/dag/dag_block_manager.cpp

```diff
@@ -221,6 +222,25 @@ DagBlockManager::InsertAndVerifyBlockReturnType DagBlockManager::verifyBlock(con
     markBlockInvalid(block_hash);
     return InsertAndVerifyBlockReturnType::NotEligible;
   }
+  {
```

Comment:
any special reason why this new part of code is under it's own block {...}?

Reply:
No. I just wanted to group this code. Maybe it is better to move it to separate method, but we will need to have additional code to process it. Like try catch above, or by returning and processing some optional value for example

Reply:
you can separate it by some comment and visually it would have the same effect:

```
// This part of code does this and that
u256 total_block_weight = 0;
const auto &trxs = blk.getTrxs();
const auto &estimations = blk.getEstimations();
if (trxs.size() != estimations.size()) {
  return InsertAndVerifyBlockReturnType::IncorrectTransactionsEstimation;
}
for (uint32_t i = 0; i < trxs.size(); ++i) {
  const auto &e = trx_mgr_->estimateTransactionByHash(trxs[i], propose_period);
  if (e != estimations[i]) {
    return InsertAndVerifyBlockReturnType::IncorrectTransactionsEstimation;
  }
  total_block_weight += estimations[i];
}
if (total_block_weight > getDagConfig().gas_limit) {
  return InsertAndVerifyBlockReturnType::BlockTooBig;
}

// And this does another thing....
```